### PR TITLE
feat: add Create Share sql support

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

feat: add Create Share sql support

Fixes #issue https://github.com/datafuselabs/databend/issues/6974

### CODE REVIEW

File: query/src/catalogs/default/database_catalog.rs [hunk 1]
```diff
@@ -460,4 +462,22 @@ impl Catalog for DatabaseCatalog {
...
+        }
+        info!("Create share from req:{:?}", req);
+
+        // if self
+        // .immutable_catalog
+        // .exists_database(req.tenant(), req.db_name())
+        // .await?
+        // {
+        // return self.immutable_catalog.create_table(req).await;
+        // }
```
Comment (leiysky) [inline query/src/catalogs/default/database_catalog.rs:480]: Do we need this?

**Review by leiysky — APPROVED** [review]

File: common/ast/src/parser/statement.rs [hunk 2]
```diff
@@ -741,6 +741,24 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
         },
     );
 
+    // share statements
+    let create_share = map(
+        rule! {
+            CREATE ~ SHARE ~ (IF ~ NOT ~ EXISTS )? ~ #peroid_separated_idents_1_to_2 ~ ( COMMENT ~ "=" ~ #literal_string)?
```
Comment (Xuanwo) [inline common/ast/src/parser/statement.rs:747]: Is this sql complete?

Can you show me how users will use this?

File: common/ast/src/parser/statement.rs [hunk 3]
```diff
@@ -741,6 +741,24 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
...
+        },
+        |(_, _, opt_if_not_exists, (catalog, share), comment_opt)| {
```
Comment (Xuanwo) [inline common/ast/src/parser/statement.rs:749]: Do we need to include the catalog in `share`? I thought we could handle them in the same way as `stage`.

Comment (lichuang) [reply common/ast/src/parser/statement.rs:749]: > Do we need to include the catalog in `share`? I thought we could handle them in the same way as `stage`.

@leiysky @dantengsky

Comment (BohuTANG) [reply common/ast/src/parser/statement.rs]: Got it:)

**Review by BohuTANG — APPROVED** [review]

File: common/ast/src/ast/statements/share.rs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    pub comment: Option<String>,
+}
+
+impl Display for CreateShareStmt<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "CREATE SHARE ")?;
+        if self.if_not_exists {
+            write!(f, "IF NOT EXISTS ")?;
+        }
+        write!(f, "{:?}", self.share)?;
```
Comment (Xuanwo) [inline common/ast/src/ast/statements/share.rs:33]:
```suggestion
        write!(f, "{}", self.share)?;
```

Print `Identifier` as debug is not expected.

File: common/ast/tests/it/testdata/statement.txt [hunk 5]
```diff
@@ -5802,3 +5802,41 @@ Presign(
 )
 
 
+---------- Input ----------
+CREATE SHARE t COMMENT='share comment';
+---------- Output ---------
+CREATE SHARE Identifier { name: "t", quote: None, span: Ident(13..14) } COMMENT = share comment
```
Comment (Xuanwo) [inline common/ast/tests/it/testdata/statement.txt:5808]: Not an expected output.

We are expecting `CREATE SHARE t COMMENT='share comment';`

File: common/ast/tests/it/testdata/statement.txt [hunk 6]
```diff
@@ -5802,3 +5802,41 @@ Presign(
...
+            "share comment",
+        ),
+    },
+)
+
+
+---------- Input ----------
+CREATE SHARE IF NOT EXISTS t;
+---------- Output ---------
+CREATE SHARE IF NOT EXISTS Identifier { name: "t", quote: None, span: Ident(27..28) }
```
Comment (Xuanwo) [inline common/ast/tests/it/testdata/statement.txt:5828]: the same.

**Review by Xuanwo — CHANGES_REQUESTED** [review]

File: common/ast/src/ast/statements/share.rs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        if let Some(comment) = &self.comment {
+            write!(f, " COMMENT = {comment}")?;
```
Comment (Xuanwo) [inline common/ast/src/ast/statements/share.rs:35]:
```suggestion
            write!(f, " COMMENT = '{comment}'")?;
```

File: common/ast/src/ast/statements/share.rs [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+use crate::ast::Identifier;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct CreateShareStmt<'a> {
+    pub if_not_exists: bool,
+    pub share: Identifier<'a>,
+    pub comment: Option<String>,
+}
+
+impl Display for CreateShareStmt<'_> {
```
Comment (Xuanwo) [inline common/ast/src/ast/statements/share.rs:27]: `Display` of `CreateShareStmt` should be the normalized SQL representation.

**Review by Xuanwo — CHANGES_REQUESTED** [review]

**Pull request merged**